src/test/java/com/fasterxml/jackson/databind/deser/TestCustomSubTypeValidator.java

```diff
@@ -0,0 +1,277 @@
+            + "      \"name\" : \"Tux\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    protected static final String PETTING_ZOO_JSON = "{"
+            + "  \"name\" : \"Jacksonville Petting Zoo\","
+            + "  \"animals\": ["
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Goat\","
+            + "      \"name\" : \"Bill\""
+            + "    },"
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Sheep\","
+            + "      \"name\" : \"Shaun\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    /*
+    /**********************************************************
+    /* Helper classes (beans)
+    /**********************************************************
+     */
+
+    public static class Zoo
+    {
+        private String _name;
+
+        private List<Animal> _animals;
+
+        public String getName()
+        {
+            return _name;
+        }
+
+        public void setName(String name)
+        {
+            _name = name;
+        }
+
+        public List<Animal> getAnimals()
+        {
+            return _animals;
+        }
+
+        public void setAnimals(List<Animal> animals)
+        {
+            _animals = animals;
+        }
+    }
+
+    @JsonTypeInfo(
+        use=JsonTypeInfo.Id.CLASS,
+        include=JsonTypeInfo.As.PROPERTY,
+        property="species"
+    )
+    public static abstract class Animal
+    {
+        private String _name;
+
+        public String getName() {
+            return _name;
+        }
+
+        public void setName(String name) {
+            _name = name;
+        }
+    }
+
+    public static class Lion extends Animal
+    {
+    }
+
+    public static class Warthog extends Animal
+    {
+    }
+
+    public static class Goat extends Animal
+    {
+    }
+
+    public static class Sheep extends Animal
+    {
+    }
+
+    public static class Penguin extends Animal
+    {
+    }
+
+    /*
+    /**********************************************************
+    /* Helper classes (validators)
+    /**********************************************************
+     */
+
+    static class DenyAllValidator
+        extends SubTypeValidator
+    {
+        @Override
+        public void validateSubType(DeserializationContext ctxt, JavaType type, BeanDescription beanDesc)
+            throws JsonMappingException 
+        {
+            ctxt.reportBadTypeDefinition(
+                    beanDesc,
+                    "Illegal type (%s) to deserialize: this validator does not allow deserialization at all!",
+                    type.getRawClass().getName());
+        }
+    }
+
+    static class FriendlyAnimalValidator
+        extends SubTypeValidator
+    {
+        private static Class<?>[] FRIENDLY_ANIMAL_WHITELIST = new Class<?>[] {
+            Animal.class,
```

Comment:
Originally, I did not plan to include `Animal` itself here. It is an abstract class, and I always specify concrete sub-types in my JSON payloads.

However, during debugging I noticed that `Animal` itself sometimes gets passed as `JavaType` into `SubTypeValidator.validateSubType(...)`. I could not figure out why yet. I fear it might be an unrelated bug in how the `DeserializerFactory` calls `SubTypeValidator`. If that were true, the existing black-list may not be effective in all situations!?! More likely, I have missed something trivial.